storage/src/vespa/storage/storageserver/changedbucketownershiphandler.cpp

```diff
@@ -167,11 +183,16 @@ class StateDiffLazyAbortPredicate
     // Fast path to avoid trying (and failing) to compute owner in a state
     // where all distributors are down.
     bool _allDistributorsHaveGoneDown;
+    uint16_t _nodeIndex;
 
     bool doShouldAbort(const document::Bucket &bucket) const override {
         if (_allDistributorsHaveGoneDown) {
             return true;
         }
+        bool storageNodeUp = _newState.storageNodeUp(bucket.getBucketSpace(), _nodeIndex);
+        if (!storageNodeUp) {
```

Comment:
Consider factoring this out to `if (!contentNodeUpInBucketSpace(bucket.getBucketSpace())` (or something like that) since the only local parameter used is `bucket`. Also easier to hide any up front caching that way.